src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php

```diff
@@ -237,17 +241,30 @@ protected function configureOptions(OptionsResolver $resolver): void
 
     private function createTranslations(ProductInterface $product, array $options): void
     {
+        // add translation for each defined locales
         foreach ($this->getLocales() as $localeCode) {
-            $product->setCurrentLocale($localeCode);
-            $product->setFallbackLocale($localeCode);
+            $this->createTranslation($product, $localeCode, $options);
+        }
 
-            $product->setName($options['name']);
-            $product->setSlug($options['slug']);
-            $product->setShortDescription($options['short_description']);
-            $product->setDescription($options['description']);
+        // create or replace with custom translations
+        foreach ($options['translations'] as $localeCode => $translationOptions) {
+            $this->createTranslation($product, $localeCode, $translationOptions);
         }
     }
 
+    private function createTranslation(ProductInterface $product, string $localeCode, array $options = []): void
+    {
+        $options = $this->optionsResolver->resolve($options);
+
+        $product->setCurrentLocale($localeCode);
+        $product->setFallbackLocale($localeCode);
+
+        $product->setName($options['name']);
```

Comment:
If I understood it correctly, there is a high possibility, that this code will throw an exception if `name`, `description` or  `short_description ` won't be defined for translation. We should ensure some structure here and default values. 

Can we also write some tests for this configuration? We could take this factory and try to create different objects for different options.

Reply:
I'm not really sure how the factory works, but I'll take a look at the taxon factory for ideas on implementing some defaults.

Reply:
By the way, `name`, `description` and `short_description` are not allowed to be empty, so the translations will not be the first to throw an exception because of those values missing.

Reply:
I have added some tests, als for incomplete translations!